src/librustc/middle/recursion_limit.rs

```diff
@@ -6,26 +6,57 @@
 // just peeks and looks for that attribute.
 
 use crate::session::Session;
+use core::num::IntErrorKind;
 use rustc_span::symbol::{sym, Symbol};
 use syntax::ast;
 
 use rustc_data_structures::sync::Once;
 
 pub fn update_limits(sess: &Session, krate: &ast::Crate) {
-    update_limit(krate, &sess.recursion_limit, sym::recursion_limit, 128);
-    update_limit(krate, &sess.type_length_limit, sym::type_length_limit, 1048576);
+    update_limit(sess, krate, &sess.recursion_limit, sym::recursion_limit, 128);
+    update_limit(sess, krate, &sess.type_length_limit, sym::type_length_limit, 1048576);
 }
 
-fn update_limit(krate: &ast::Crate, limit: &Once<usize>, name: Symbol, default: usize) {
+fn update_limit(sess: &Session, krate: &ast::Crate, limit: &Once<usize>, name: Symbol, default: usize) {
+    
     for attr in &krate.attrs {
         if !attr.check_name(name) {
             continue;
         }
-
+        
+        
         if let Some(s) = attr.value_str() {
-            if let Some(n) = s.as_str().parse().ok() {
-                limit.set(n);
-                return;
+            match s.as_str().parse() {
+                Ok(n) => {
+                    limit.set(n);
+                    return;
+                }
+                Err(e) => {
+                    let mut err = sess.struct_span_err(attr.span, "recursion_limit parsing");
+                    
+                    match e.kind() {
+                        IntErrorKind::Overflow => {
+                            err.span_label(attr.span, "the recursion_limit overflowed a `usize`");
+                        }
+                        IntErrorKind::Empty => {
+                            err.span_label(attr.span, "the recursion_limit must be a positive integer");
+                        }
+                        IntErrorKind::Underflow => {
+                            // A negative number will be caught by the invalid digit handler.
```

Comment:
Would you like me to explicitly handle this case? Though I cannot seem to test it.

Reply:
Hmm, would it hurt to explicitly handle it? If yes, you could use the [`bug!` macro](https://doc.rust-lang.org/nightly/nightly-rustc/rustc/macro.bug.html).

Reply:
I would use `bug!` here, as @hellow554 suggests.